File: generator/generator.py
import random, sys

class Generator:
    randomizer = random.Random(int(random.uniform(0, sys.maxsize)))
    
    def __init__(self, n, k):
        self.n = n
        self.k = k
    
    def gen(self):
        mx = Generator.randomizer.randint(self.k, sys.maxsize)
        mn = mx // self.k
        matrix = [0] * self.n
        matrix[0] = mn
        matrix[-1] = mx
        for i in range(1, self.n - 1):
            matrix[i] = Generator.randomizer.randint(mn + 1, mx - 1)
        return matrix

def gen(n, k):
    return Generator(n, k).gen()

File: lib/Generator.java
import java.util.Random;

public final class Generator {
	private final static Random RANDOMIZER = new Random((int) (Integer.MAX_VALUE * Math.random()));
	private final int n, k;

	public Generator(final int n, final int k) {
		this.n = n;
		this.k = k;
	}

	public int[] gen() {
		final int mx = RANDOMIZER.nextInt(k, Integer.MAX_VALUE);
		final int mn = (mx / k);
		final int[] matrix = new int[n];
		matrix[0] = mn;
		matrix[n - 1] = mx;
		for (int i = 1; i < n - 1; i++) {
			matrix[i] = RANDOMIZER.nextInt(mn + 1, mx - 1);
		}
		return matrix;
	}
}
